table/table.go

```diff
@@ -127,24 +145,24 @@ type Table interface {
 	// AllocAutoID allocates an auto_increment ID for a new row.
 	AllocAutoID(ctx sessionctx.Context) (int64, error)
 
-	// Allocator returns Allocator.
-	Allocator(ctx sessionctx.Context) autoid.Allocator
-
 	// RebaseAutoID rebases the auto_increment ID base.
 	// If allocIDs is true, it will allocate some IDs and save to the cache.
 	// If allocIDs is false, it will not allocate IDs.
 	RebaseAutoID(ctx sessionctx.Context, newBase int64, allocIDs bool) error
 
-	// Meta returns TableInfo.
-	Meta() *model.TableInfo
-
 	// Seek returns the handle greater or equal to h.
 	Seek(ctx sessionctx.Context, h int64) (handle int64, found bool, err error)
 
 	// Type returns the type of table
 	Type() Type
 }
 
+// PartitionTable inherits Table and it's a real Table, rather than a partition.
+type PartitionTable interface {
+	Table
+	GetPartition(pid int64) Table
```

Comment:
How about defining `Partition` as an alias of `Table`.

Reply:
You mean here? https://github.com/pingcap/tidb/pull/6963/files#diff-c709fb799ac0eac6d7ea40b2278c90d7R43